contracts/0.4.24/nos/NodeOperatorsRegistry.sol

```diff
@@ -1,702 +1,1169 @@
-    bytes32 constant public ADD_NODE_OPERATOR_ROLE = keccak256("ADD_NODE_OPERATOR_ROLE");
-    bytes32 constant public SET_NODE_OPERATOR_ACTIVE_ROLE = keccak256("SET_NODE_OPERATOR_ACTIVE_ROLE");
-    bytes32 constant public SET_NODE_OPERATOR_NAME_ROLE = keccak256("SET_NODE_OPERATOR_NAME_ROLE");
-    bytes32 constant public SET_NODE_OPERATOR_ADDRESS_ROLE = keccak256("SET_NODE_OPERATOR_ADDRESS_ROLE");
-    bytes32 constant public SET_NODE_OPERATOR_LIMIT_ROLE = keccak256("SET_NODE_OPERATOR_LIMIT_ROLE");
-    bytes32 constant public REPORT_STOPPED_VALIDATORS_ROLE = keccak256("REPORT_STOPPED_VALIDATORS_ROLE");
+    /// @dev link to the index of operations with keys
+    // bytes32 internal constant KEYS_OP_INDEX_POSITION = keccak256("lido.NodeOperatorsRegistry.keysOpIndex");
+    bytes32 internal constant KEYS_OP_INDEX_POSITION = 0xcd91478ac3f2620f0776eacb9c24123a214bcb23c32ae7d28278aa846c8c380e;
 
-    uint256 constant public PUBKEY_LENGTH = 48;
-    uint256 constant public SIGNATURE_LENGTH = 96;
-    uint256 constant public MAX_NODE_OPERATORS_COUNT = 200;
+    /// @dev module type
+    // bytes32 internal constant TYPE_POSITION = keccak256("lido.NodeOperatorsRegistry.type");
+    bytes32 internal constant TYPE_POSITION = 0xbacf4236659a602d72c631ba0b0d67ec320aaf523f3ae3590d7faee4f42351d0;
 
-    uint256 internal constant UINT64_MAX = uint256(uint64(-1));
+    // bytes32 internal constant TYPE_POSITION = keccak256("lido.NodeOperatorsRegistry.stuckPenaltyDelay");
+    bytes32 internal constant STUCK_PENALTY_DELAY_POSITION = 0x8e3a1f3826a82c1116044b334cae49f3c3d12c3866a1c4b18af461e12e58a18e;
 
-    bytes32 internal constant SIGNING_KEYS_MAPPING_NAME = keccak256("lido.NodeOperatorsRegistry.signingKeysMappingName");
+    //
+    // DATA TYPES
+    //
 
     /// @dev Node Operator parameters and internal state
     struct NodeOperator {
-        bool active;    // a flag indicating if the operator can participate in further staking and reward distribution
-        address rewardAddress;  // Ethereum 1 address which receives steth rewards for this operator
-        string name;    // human-readable name
-        uint64 stakingLimit;    // the maximum number of validators to stake for this operator
-        uint64 stoppedValidators;   // number of signing keys which stopped validation (e.g. were slashed)
-
-        uint64 totalSigningKeys;    // total amount of signing keys of this operator
-        uint64 usedSigningKeys;     // number of signing keys of this operator which were used in deposits to the Ethereum 2
+        /// @dev Flag indicating if the operator can participate in further staking and reward distribution
+        bool active;
+        /// @dev Ethereum address on Execution Layer which receives steth rewards for this operator
+        address rewardAddress;
+        /// @dev Human-readable name
+        string name;
+        /// @dev The below variables store the signing keys info of the node operator.
```

Comment:
maybe this comment should be updated (unclear connection with the fields below)

Reply:
add more info abput signingkeys